archinstall/__init__.py

```diff
@@ -280,8 +282,20 @@ def plugin(f, *args, **kwargs):
 
 def _check_new_version():
 	info("Checking version...")
-	Pacman.run("-Sy")
-	upgrade = Pacman.run("-Qu archinstall").decode()
+
+	try:
+		Pacman.run("-Sy")
+	except Exception as e:
+		debug(f'Failed to perform version check: {e}')
+		info(f'Arch Linux mirrors are not reachable. Please check your internet connection')
+		exit(1)
+
+	upgrade = None
```

Comment:
Could delete this line and add an else statement to the try block below with the `upgrade` conditional inside it.